s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -619,3 +635,61 @@ func (t *TidbClient) GetBucketVersioning(ctx context.Context, bucketName string)
 	}
 	return
 }
+
+func (t *TidbClient) CreateBucketSSE(ctx context.Context, bucketName string, sseType string) error {
+	log.Infof("create bucket[%s] SSE info[%s] into tidb ...\n", bucketName, sseType)
+
+	sql := "insert into bucket_sseopts(bucketname, sse) values(?,?);"
+	args := []interface{}{bucketName, sseType}
+
+	_, err := t.Client.Exec(sql, args...)
+	if err != nil {
+		return handleDBError(err)
+	}
+
+	return nil
+}
+
+func (t *TidbClient) GetBucketSSE(ctx context.Context, bucketName string) (sseOptsPtr *pb.ServerSideEncryption, err error) {
+	log.Info("list bucket SSE info from tidb ...")
+
+	var rows *sql.Rows
+	sqltext := "select sse from bucket_sseopts where bucketname=?;"
+
+	rows, err = t.Client.Query(sqltext, bucketName)
+
+	if err == sql.ErrNoRows {
+		err = nil
+		return
+	} else if err != nil {
+		err = handleDBError(err)
+		return
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		tmp := &pb.ServerSideEncryption{}
+		err = rows.Scan(
+			&tmp.SseType)
+		if err != nil {
+			err = handleDBError(err)
+			return
+		}
+		return tmp, nil
+	}
+	return
```

Comment:
before return, we should check rows.Error()

Reply:
added